### DISCUSSION TITLE
Incompatible library version

### DISCUSSION THREAD

**Discussion by orallo** [shivammathur/homebrew-php#240] [Q&A / Help]
Hi,

I've installed php (5.6 through 8.0) following the tutorial on https://getgrav.org/blog/macos-bigsur-apache-multiple-php-versions and when I try to start apache I get the following error:

`httpd: Syntax error on line 189 of /usr/local/etc/httpd/httpd.conf: Cannot load /usr/local/opt/php@7.4/lib/httpd/modules/libphp7.so into server: dlopen(/usr/local/opt/php@7.4/lib/httpd/modules/libphp7.so, 10): Library not loaded: /usr/local/opt/curl/lib/libcurl.4.dylib\n  Referenced from: /usr/local/opt/php@7.4/lib/httpd/modules/libphp7.so\n  Reason: Incompatible library version: libphp7.so requires version 12.0.0 or later, but libcurl.4.dylib provides version 11.0.0`

I'm running MacOS BigSur 11.2.3, curl 7.64.1, I have updated and upgraded all packages through brew and everything is up to data using the latest version of all packages. I've tried reinstalling curl, curl-openssl, and everything I could think of that was related to this error.

I've even tried uninstalling all php packages, curl, python, libxml2, apache and everything related and reinstalling them again to no avail.

Any and all help would be appreciated,

Thanks in advance!
-Orallo

**Comment by shivammathur** [comment]
curl `7.64.1` was released in march 2019, please update to the latest curl.

Follow these steps:
- Update Xcode CLI tools
```zsh
sudo rm -rf /Library/Developer/CommandLineTools
xcode-select --install
```

- Update brew to get the latest formulae
```zsh
brew update
```
- If you have `curl-openssl` installed, remove it as it was renamed to curl.
```
brew uninstall --ignore-dependencies curl-openssl
```

- Reinstall curl
```zsh
brew reinstall curl
```

Now `libphp7.so` should work correctly.

**Reply by orallo** [reply]
Thank you so much for your reply.

Unfortunately it hasn't worked, for me.  I'm going to try to uninstall the whole xcode suite and see if that does anything.

I will post the result as soon as I'm done.

Thanks again!
-Orallo

**Reply by orallo** [reply]
I ended up having to uninstall homebrew completely and reinstalling it from scratch.

Apparently my hombrew installation got corrupted somehow and nothing I tried would allow me to update curl to the latest version.

Thanks a lot for your help though,
-Orallo
